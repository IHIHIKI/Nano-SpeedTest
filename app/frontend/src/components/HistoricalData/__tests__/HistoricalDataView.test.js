import React from 'react';
import renderer from 'react-test-renderer';
import HistoricaDataView from '../HistoricalDataView';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';

const mockStore = configureStore();
let store = mockStore({
    ads: {
        currentAd: {
            title: 'Title',
            message: 'Message',
            url: 'url'
        }
    },
    table: {
        0: {
            id: 440,
            transactionHashSending: "A20FE63E0C55592C5EEF4089B1D66EF8B0A76DECC8F852B6BE7F112A13943AA2",
            transactionHashReceiving: "4E3D4B7415F282608CC2EAFF0120AEDBCB50174478744916F4048AECC58F49FB",
            amount: "0.0006",
            origin: {
                id: 4,
                nodeLocation: "New York",
                latitude: "40.804300",
                longitude: "-74.012100",
            },
            destination: {
                id: 3,
                nodeLocation: "Singapore",
                latitude: "1.293100",
                longitude: "103.850100"
            },
            startSendTimestamp: 1542417085995,
            endSendTimestamp: 1542417086000,
            startReceiveTimestamp: 1542417112098,
            endReceiveTimestamp: 1542417112000,
            completed: true
        }
    },
    transactions: {
        isFetchingTransaction: false,
        isFetchingTiming: false
    },
    pastResults: {
        pastTransactions:[]
    },
    nodes: [],
    form: {}
});

const props = {
    pastTransactions: [],
    totalTransactions: 1,
    globalAverage: 1,
    nodeLocations: {}
};

it('shallow renders without crashing', () => {
    shallow(<HistoricaDataView store={store} {...props}/>);
});

it(`renders base snapshot correctly`, () => {
    const tree = renderer.create(
        <Provider store={store}>
            <BrowserRouter>
                <HistoricaDataView {...props}/>
            </BrowserRouter>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
