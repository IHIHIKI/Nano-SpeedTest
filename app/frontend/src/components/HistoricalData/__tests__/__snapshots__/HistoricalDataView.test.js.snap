// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders base snapshot correctly 1`] = `
<div
  className="HistoricalData"
>
  <div
    className="nav nav-tabs"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link "
        href="/"
        onClick={[Function]}
      >
         
        Speed Test
         
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link "
        href="/Results"
        onClick={[Function]}
      >
         
        Your Results
         
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link active"
        href="/Stats"
        onClick={[Function]}
      >
         
        Global Statistics
         
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link "
        href="/Info"
        onClick={[Function]}
      >
         
        More Info
         
      </a>
    </li>
  </div>
  <div
    className="container"
  >
    <div
      className="jumbotron"
    >
      <div
        id="ad-content"
      >
        <a
          className="row"
          href="https://nano21.net"
        >
          <b>
            ♠ Nano21.net 
          </b>
            | 20 Nano Give-Away for testers! | Nano Blackjack Casino
        </a>
        <p
          className="row"
        >
          COMMUNITY AD
        </p>
      </div>
    </div>
  </div>
  <div
    className="container-fluid"
  >
    <div
      className="row form-group"
    >
      <h2
        className="map-header text-center"
      >
        Some Stats
      </h2>
      <div
        className="container-fluid"
      >
        <div
          className="row form-group"
        >
          <div
            className="col-auto"
          >
            <h3
              className="map-header"
            >
              Total Transactions Sent: 
              0
            </h3>
          </div>
          <div
            className="col-auto"
          >
            <h3
              className="map-header"
            >
              Average Time: 
              0.000
               Seconds
            </h3>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row form-group mt-5"
    >
      <div
        className="col-6"
      >
        <h2
          className="map-header text-center"
        >
          Scatter Chart of Recent Transactions
        </h2>
      </div>
      <div
        className="col-6"
      >
        <h2
          className="map-header text-center"
        >
          Locations of Our Nano Nodes
        </h2>
      </div>
    </div>
    <div
      className="row form-groups mt-5"
    >
      <div
        className="col"
      >
        <div
          className="nano-container"
        >
          <h2
            className="map-header text-left"
          >
            Past Transactions
          </h2>
          <table
            className="table"
          >
            <thead>
              <tr>
                <th
                  scope="col"
                >
                  Origin
                </th>
                <th
                  scope="col"
                >
                  Destination
                </th>
                <th
                  scope="col"
                >
                  Amount
                </th>
                <th
                  scope="col"
                >
                  Elapsed Time
                </th>
                <th
                  scope="col"
                >
                  Date
                </th>
                <th
                  className="block-col"
                  scope="col"
                >
                  Sending Block
                </th>
                <th
                  className="block-col"
                  scope="col"
                >
                  Receiving Block
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
          <div
            className="row"
          >
            <div
              className="col-sm-auto"
            >
              Showing 
              1
               to 
              0
               of 
              0
               transactions.
            </div>
            <div
              className="col-sm-auto"
            >
              <nav
                aria-label="Table navigation"
              >
                <ul
                  className="pagination"
                >
                  <li
                    className="page-item disabled"
                  >
                    <button
                      className="page-link"
                      onClick={[Function]}
                    >
                      Prev
                    </button>
                  </li>
                  <li
                    className="page-item disabled"
                  >
                    <button
                      className="page-link"
                      onClick={[Function]}
                    >
                      Next
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
